src/fq_default/ctx.c

```diff
@@ -200,21 +200,42 @@ void fq_default_ctx_modulus(fmpz_mod_poly_t p, const fq_default_ctx_t ctx)
     }
 }
 
-/* FIXME: fq_ctx_init_randtest only tests nmod sized primes, so that's
-   all we'll do here too? */
-void fq_default_ctx_init_randtest(fq_default_ctx_t ctx, flint_rand_t state, int type)
+/* Create a random fq_default_ctx picked from the five internal representations */
+void fq_default_ctx_init_randtest(fq_default_ctx_t ctx, flint_rand_t state)
 {
-    ulong prime;
-    slong degree;
-    fmpz_t p;
-
-    /* select a prime and degree to create the finite field */
-    prime = _nmod_poly_conway_rand(&degree, state, type);
-
-    /* fq_default initialisation wants an fmpz_t for the prime */
-    fmpz_init(p);
-    fmpz_set_ui(p, prime);
-
-    /* Initialise the context using the prime and degree selected above */
-    fq_default_ctx_init(ctx, p, degree, "x");
+    fmpz_t prime;
+    slong deg;
+    
+    /* Select a context type [1,...,5] */
+    int ctx_type = 1 + n_randint(state, 5);
+    switch (ctx_type)
+    {
+        /* Create GF(p^d) for FQ_ZECH context */
+        case FQ_DEFAULT_FQ_ZECH:
+            fmpz_randprime(prime, state, 2 + n_randint(state, 4), 1);
+            deg = 1 + n_randint(state, 4);
```

Comment:
It looks like this can generate GF(31^4) which is quite large for the Zech representation as far as basic testing goes anyway. Maybe drop both the ``n_randint(state, 4)`` to ``n_randint(state, 3)``.

Reply:
I have modified these two values.